awx/main/models/ha.py

```diff
@@ -176,6 +176,14 @@ class InstanceGroup(HasPolicyEditsMixin, BaseModel, RelatedJobsMixin):
         null=True,
         on_delete=models.CASCADE
     )
+    credential = models.ForeignKey(
+        'Credential',
+        related_name='%(class)ss',
+        blank=True,
+        null=True,
+        default=None,
+        on_delete=models.SET_NULL,
```

Comment:
can this ever be edited by a user? If not, I prefer to set `editable=False`. Also, this should be exposed in the serializer so that people can see what credential is linked via the API. If you just add it to the fields, then it will present itself as a read-only field like this.

Reply:
@AlanCoding by 'this' do you mean the ability to modify the associated credential? If that's what you meant, then yes they will be able to. [Here](https://tower-mockups.testing.ansible.com/mockups/instance_groups/instance-groups-container-add/) is how this will be surfaced in the UI. InstanceGroups will only have credentials in the context of this new "Container Group" concept. We're still wrestling with the naming here so if you have any thoughts I'd love to hear them.

I should mention that I just copy / pasted this association off of another model, so we should definitely make sure I'm doing the right thing here. 😄 As for exposing through the API - absolutely. I have mostly been working in the task manager up until now but I will be looking to do this soon and may have a question or two for you.